test/test_trixi.jl

```diff
@@ -29,44 +29,71 @@ function test_trixi_run(parameters_file; l2=nothing, linf=nothing, atol=200*eps(
 end
 
 
+# Get the first value assigned to `keyword` in `args` and return `default_value`
+# if there are no assignments to `keyword` in `args`.
+function get_kwarg(args, keyword, default_value)
+  val = default_value
+  for arg in args
+    if arg.head == :(=) && arg.args[1] == keyword
+      val = arg.args[2]
+      break
+    end
+  end
+  return val
+end
+
+# Use a macro to avoid world age issues when defining new initial conditions etc.
+# inside an elixir.
 """
-    test_trixi_include(parameters_file; l2=nothing, linf=nothing, atol=10*eps(), rtol=0.001, parameters...)
+    test_trixi_include(elixir; l2=nothing, linf=nothing,
+                               atol=10*eps(), rtol=0.001,
+                               parameters...)
 
-Test Trixi by calling `trixi_include(parameters_file; parameters...)`.
+Test Trixi by calling `trixi_include(elixir; parameters...)`.
 By default, only the absence of error output is checked.
 If `l2` or `linf` are specified, in addition the resulting L2/Linf errors
 are compared approximately against these reference values, using `atol, rtol`
 as absolute/relative tolerance.
 """
-function test_trixi_include(parameters_file; l2=nothing, linf=nothing,
-                                             atol=200*eps(), rtol=0.001,
-                                             kwargs...)
+macro test_trixi_include(elixir, args...)
 
-  println("#"^80)
-  println(parameters_file)
+  local l2   = get_kwarg(args, :l2, nothing)
+  local linf = get_kwarg(args, :linf, nothing)
+  local atol = get_kwarg(args, :atol, 200*eps())
+  local rtol = get_kwarg(args, :rtol, 0.001)
+  local kwargs = Pair{Symbol, Any}[]
+  for arg in args
+    if arg.head == :(=) && !(arg.args[1] in (:l2, :linf, :atol, :rtol))
+      push!(kwargs, Pair(arg.args...))
```

Comment:
Out of curiosity: wouldn't `arg.args` already have to be a `Pair{Symbol, Any}` and you could just do
```suggestion
      push!(kwargs, arg.args)
```

Reply:
No. It's a `Vector{Any}` containing a `Symbol` and an `Expr`. For example,
```julia
julia> @test_trixi_include("examples/2d/elixir_advection_basic.jl", l2=[9.144681765639205e-6], linf=[6.437440532547356e-5], tspan = (0.0, 0.1))
```
will result in `arg.args` of the form
```julia
Array{Any}((2,))
  1: Symbol tspan
  2: Expr
    head: Symbol tuple
    args: Array{Any}((2,))
      1: Float64 0.0
      2: Float64 0.1
```

Reply:
Thanks for clearing this up for me!